src/instructlab/sdg/pipeline.py

```diff
@@ -20,6 +20,8 @@ def __init__(
         self.model_id = model_id
         self.num_instructions_to_generate = num_instructions_to_generate
         self.sdg_base = resources.files(__package__)
+        # FIXME: base this on the available number of CPUs
+        self.num_procs = 8
```

Comment:
It would be great to make this a parameter.  `ilab data generate` already exposes this as a tunable thing.

```
$ ilab data generate --help
...
  --num-cpus INTEGER          Number of processes to use.  [default: 10]
...
```

It defaults to 10, though it should be fixed to default to the number of CPUs available. Either way, at least for our `ilab` use case, we have this passed in to `generate_data()` but we're not using it right now.